test/view.js

```diff
@@ -0,0 +1,397 @@
+      model.add('fruits', {name: 'grapefruit', color: 'orange', amount: 10, id: 'grapefruitId'});
+      var expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+    });  
+
+    it('non-empty colection: adds to view as conditions ARE satisfied', function() {
+      var model    = (new Model).at('_page');
+
+      model = addData(model); // Add items to collection
+      model.fn('yellowFruits', function(emit, fruit) {
+        if (fruit.color === 'yellow') {
+          emit(fruit.name + '*' + fruit.color + '*' + fruit.amount*2, '_page.fruits.' + fruit.id);
+        }
+      });
+      var view = model.at('fruits').view('yellowFruits');
+      view.ref('_page.filteredFruits');
+
+      // Add new item to non-empty collection
+      model.add('fruits', {name: 'mango', color: 'yellow', amount: 15, id: 'mangoId'});      
+      var expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId', 'mangoId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+    });
+  });
+
+  describe('Removing item from collection', function() {
+    it('removes item from view as it was included', function() {
+      var model    = (new Model).at('_page');
+  
+      model = addData(model); // Add items to collection
+      model.fn('yellowFruits', function(emit, fruit) {
+        if (fruit.color === 'yellow') {
+          emit(fruit.name + '*' + fruit.color + '*' + fruit.amount*2, '_page.fruits.' + fruit.id);
+        }
+      });
+
+      // Before removing included item
+      var view = model.at('fruits').view('yellowFruits');
+      view.ref('_page.filteredFruits');
+      var expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+      
+      model.del('fruits.bananaId'); // Remove included item
+            
+      // After removing included item
+      expectedFruits = createExpectedResult(model, ['lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+    });
+
+    it('does NOT make any changes to view as item was NOT included', function() {
+      var model    = (new Model).at('_page');
+
+      model = addData(model); // Add items to collection
+      model.fn('yellowFruits', function(emit, fruit) {
+        if (fruit.color === 'yellow') {
+          emit(fruit.name + '*' + fruit.color + '*' + fruit.amount*2, '_page.fruits.' + fruit.id);
+        }
+      });
+
+      // Before removing item NOT included in view
+      var view = model.at('fruits').view('yellowFruits');
+      view.ref('_page.filteredFruits');
+      var expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+
+      model.del('fruits.appleId'); // Remove item
+
+      // After removing item NOT included in view
+      expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId']);
```

Comment:
No need to create expectedFruits again since you're creating an exact copy of it from before.